osmoutils/cosmwasm/helpers_test.go

```diff
@@ -0,0 +1,109 @@
+package cosmwasm_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/osmosis-labs/osmosis/osmoutils/cosmwasm"
+	"github.com/osmosis-labs/osmosis/v23/app/apptesting"
+)
+
+type KeeperTestSuite struct {
+	apptesting.KeeperTestHelper
+}
+
+func TestKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(KeeperTestSuite))
+}
+
+func (s *KeeperTestSuite) TestSudoGasLimit() {
+	// Skip test if there is system-side incompatibility
+	s.SkipIfWSL()
+
+	// We use contracts already defined in existing modules to avoid duplicate test contract code.
+	// This is a simple counter contract that counts `Amount` times and does a state write on each iteration.
+	// Source code can be found in x/concentrated-liquidity/testcontracts/contract-sources
+	counterContractPath := "../../x/concentrated-liquidity/testcontracts/compiled-wasm/counter.wasm"
+
+	// Message structs for the test CW contract
+	type CountMsg struct {
+		Amount int64 `json:"amount"`
+	}
+	type CountMsgResponse struct {
+	}
+	type CountSudoMsg struct {
+		Count CountMsg `json:"count"`
+	}
+
+	tests := map[string]struct {
+		wasmFile      string
+		msg           CountSudoMsg
+		noContractSet bool
+
+		expectedError error
+	}{
+		"contract consumes less than limit": {
+			wasmFile: counterContractPath,
+			msg: CountSudoMsg{
+				Count: CountMsg{
+					// Consumes roughly 100k gas, which should be comfortably under the limit.
+					Amount: 10,
+				},
+			},
+		},
+		"contract that consumes more than limit": {
+			wasmFile: counterContractPath,
+			msg: CountSudoMsg{
+				Count: CountMsg{
+					// Consumes roughly 1B gas, which is well above the 30M limit.
+					Amount: 100000,
+				},
+			},
+			expectedError: fmt.Errorf("contract call ran out of gas"),
+		},
+	}
+	for name, tc := range tests {
+		s.Run(name, func() {
+			s.Setup()
+
+			// We use a gov permissioned contract keeper to avoid having to manually set permissions
+			contractKeeper := wasmkeeper.NewGovPermissionKeeper(s.App.WasmKeeper)
+
+			// Upload and instantiate wasm code
+			_, cosmwasmAddressBech32 := s.uploadAndInstantiateContract(contractKeeper, tc.wasmFile)
+
+			// System under test
+			response, err := cosmwasm.Sudo[CountSudoMsg, CountMsgResponse](s.Ctx, contractKeeper, cosmwasmAddressBech32, tc.msg)
+
+			fmt.Println("response: ", response)
+
+			if tc.expectedError != nil {
+				s.Require().ErrorContains(err, tc.expectedError.Error())
+				return
+			}
+
+			s.Require().NoError(err)
```

Comment:
Let's also check the response

Reply:
The test contract is intended to just check gas so it doesn't return anything substantive, but I did add a response check here 114ca628b25d4806d2262eb8965aecf4680270e9